src/swtpm_setup/swtpm.c

```diff
@@ -67,16 +67,19 @@ static int swtpm_start(struct swtpm *self)
     gboolean success;
     GError *error = NULL;
     unsigned ctr;
+    int pidfile_fd;
+    int ret = 1;
 
-    self->pidfile = g_strjoin(G_DIR_SEPARATOR_S,
-                              self->state_path, ".swtpm_setup.pidfile", NULL);
-    pidfile_file = g_strdup_printf("file=%s", self->pidfile);
+    g_free(self->pidfile);
+    self->pidfile = g_strdup(".swtpm_setup.pidfile.XXXXXX");
+    pidfile_fd = mkstemp(self->pidfile);
```

Comment:
Do we still need `self->pidfile` at all then since this is now just a template.

Reply:
Na, you are right. It will get the actual filename then.

Reply:
Hm, I kept it so it can be unlinked in 'cleanup', but that could be done locally in the function as well (since the fd is already open)... I can change it if you want.